In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

In [2]:
df = pd.read_csv('previous_application.csv')
pd.set_option('display.max_columns', None)

In [3]:
df.head()

SK_ID_PREV  SK_ID_CURR NAME_CONTRACT_TYPE  AMT_ANNUITY  AMT_APPLICATION  \
0     2030495      271877     Consumer loans     1730.430          17145.0   
1     2802425      108129         Cash loans    25188.615         607500.0   
2     2523466      122040         Cash loans    15060.735         112500.0   
3     2819243      176158         Cash loans    47041.335         450000.0   
4     1784265      202054         Cash loans    31924.395         337500.0   

   AMT_CREDIT  AMT_DOWN_PAYMENT  AMT_GOODS_PRICE WEEKDAY_APPR_PROCESS_START  \
0     17145.0               0.0          17145.0                   SATURDAY   
1    679671.0               NaN         607500.0                   THURSDAY   
2    136444.5               NaN         112500.0                    TUESDAY   
3    470790.0               NaN         450000.0                     MONDAY   
4    404055.0               NaN         337500.0                   THURSDAY   

   HOUR_APPR_PROCESS_START FLAG_LAST_APPL_PER_CONTRACT  \
0                       15                           Y   
1                       11                           Y   
2                       11                           Y   
3                        7                           Y   
4                        9                           Y   

   NFLAG_LAST_APPL_IN_DAY  RATE_DOWN_PAYMENT  RATE_INTEREST_PRIMARY  \
0                       1                0.0               0.182832   
1                       1                NaN                    NaN   
2                       1                NaN                    NaN   
3                       1                NaN                    NaN   
4                       1                NaN                    NaN   

   RATE_INTEREST_PRIVILEGED NAME_CASH_LOAN_PURPOSE NAME_CONTRACT_STATUS  \
0                  0.867336                    XAP             Approved   
1                       NaN                    XNA             Approved   
2                       NaN                    XNA             Approved   
3                       NaN                    XNA             Approved   
4                       NaN                Repairs              Refused   

   DAYS_DECISION      NAME_PAYMENT_TYPE CODE_REJECT_REASON  NAME_TYPE_SUITE  \
0            -73  Cash through the bank                XAP              NaN   
1           -164                    XNA                XAP    Unaccompanied   
2           -301  Cash through the bank                XAP  Spouse, partner   
3           -512  Cash through the bank                XAP              NaN   
4           -781  Cash through the bank                 HC              NaN   

  NAME_CLIENT_TYPE NAME_GOODS_CATEGORY NAME_PORTFOLIO NAME_PRODUCT_TYPE  \
0         Repeater              Mobile            POS               XNA   
1         Repeater                 XNA           Cash            x-sell   
2         Repeater                 XNA           Cash            x-sell   
3         Repeater                 XNA           Cash            x-sell   
4         Repeater                 XNA           Cash           walk-in   

              CHANNEL_TYPE  SELLERPLACE_AREA NAME_SELLER_INDUSTRY  \
0             Country-wide                35         Connectivity   
1           Contact center                -1                  XNA   
2  Credit and cash offices                -1                  XNA   
3  Credit and cash offices                -1                  XNA   
4  Credit and cash offices                -1                  XNA   

   CNT_PAYMENT NAME_YIELD_GROUP       PRODUCT_COMBINATION  DAYS_FIRST_DRAWING  \
0         12.0           middle  POS mobile with interest            365243.0   
1         36.0       low_action          Cash X-Sell: low            365243.0   
2         12.0             high         Cash X-Sell: high            365243.0   
3         12.0           middle       Cash X-Sell: middle            365243.0   
4         24.0             high         Cash Street: high                 NaN   

   D

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1670214 entries, 0 to 1670213
Data columns (total 37 columns):
 #   Column                       Non-Null Count    Dtype  
---  ------                       --------------    -----  
 0   SK_ID_PREV                   1670214 non-null  int64  
 1   SK_ID_CURR                   1670214 non-null  int64  
 2   NAME_CONTRACT_TYPE           1670214 non-null  object 
 3   AMT_ANNUITY                  1297979 non-null  float64
 4   AMT_APPLICATION              1670214 non-null  float64
 5   AMT_CREDIT                   1670213 non-null  float64
 6   AMT_DOWN_PAYMENT             774370 non-null   float64
 7   AMT_GOODS_PRICE              1284699 non-null  float64
 8   WEEKDAY_APPR_PROCESS_START   1670214 non-null  object 
 9   HOUR_APPR_PROCESS_START      1670214 non-null  int64  
 10  FLAG_LAST_APPL_PER_CONTRACT  1670214 non-null  object 
 11  NFLAG_LAST_APPL_IN_DAY       1670214 non-null  int64  
 12  RATE_DOWN_PAYMENT            774370 non-nu

In [5]:
pd.set_option('display.max_columns', None)

# Pilih kolom-kolom bertipe objek dan simpan dalam variabel 'categoric'
categoric = df.select_dtypes('object')

# Pilih kolom-kolom bertipe numerik dan simpan dalam variabel 'numeric'
numeric = df.select_dtypes('number')

# Tentukan jumlah baris dan kolom
rows = df.shape[0]
cols = df.shape[1]

# Cetak bentuk DataFrame
print(f'Jumlah Baris: {rows}\nJumlah Kolom: {cols}')

# Cetak sampel 6 baris dari DataFrame
print('\nContoh Data:')
display(df.sample(6))

# Cetak ringkasan statistik data kategorikal
print('\nRingkasan Statistik Data Kategorikal:')
display(categoric.describe().T)

# Cetak ringkasan statistik data numerik
print('\nRingkasan Statistik Data Numerik:')
display(numeric.describe().T)

# Define categorical_col and numerical_col before printing
categorical_col = categoric.columns.tolist()  # Get the list of categorical columns
numerical_col = numeric.columns.tolist()    # Get the list of numerical columns

Jumlah Baris: 1670214
Jumlah Kolom: 37

Contoh Data:


SK_ID_PREV  SK_ID_CURR NAME_CONTRACT_TYPE  AMT_ANNUITY  \
886608      2359715      383742         Cash loans          NaN   
1214692     2131694      453709         Cash loans    34117.155   
1105368     2303636      341447     Consumer loans     3292.380   
768449      1493885      230930         Cash loans    16825.500   
419154      2651240      277522         Cash loans    12946.725   
827004      1037058      365942         Cash loans          NaN   

         AMT_APPLICATION  AMT_CREDIT  AMT_DOWN_PAYMENT  AMT_GOODS_PRICE  \
886608               0.0         0.0               NaN              NaN   
1214692        1125000.0   1288350.0               NaN        1125000.0   
1105368          16645.5     17635.5               0.0          16645.5   
768449          450000.0    450000.0               NaN         450000.0   
419154          148500.0    163201.5               NaN         148500.0   
827004               0.0         0.0               NaN              NaN   

        WEEKDAY_APPR_PROCESS_START  HOUR_APPR_PROCESS_START  \
886608                      MONDAY                       13   
1214692                    TUESDAY                       11   
1105368                     MONDAY                       15   
768449                    SATURDAY                       13   
419154                     TUESDAY                       10   
827004                    SATURDAY                       15   

        FLAG_LAST_APPL_PER_CONTRACT  NFLAG_LAST_APPL_IN_DAY  \
886608                            Y                       1   
1214692                           Y                       1   
1105368                           Y                       1   
768449                            Y                       1   
419154                            Y                       1   
827004                            Y                       1   

         RATE_DOWN_PAYMENT  RATE_INTEREST_PRIMARY  RATE_INTEREST_PRIVILEGED  \
886608                 NaN                    NaN                       NaN   
1214692                NaN                    NaN                       NaN   
1105368                0.0                    NaN                       NaN   
768449                 NaN                    NaN                       NaN   
419154                 NaN                    NaN                       NaN   
827004                 NaN                    NaN                       NaN   

        NAME_CASH_LOAN_PURPOSE NAME_CONTRACT_STATUS  DAYS_DECISION  \
886608                     XNA             Canceled           -196   
1214692                Repairs              Refused           -825   
1105368                    XAP             Approved           -575   
768449                     XNA             Approved           -404   
419154                     XNA              Refused           -375   
827004                     XNA             Canceled           -335   

             NAME_PAYMENT_TYPE CODE_REJECT_REASON NAME_TYPE_SUITE  \
886608                     XNA                XAP             NaN   
1214692                    XNA              LIMIT             NaN   
1105368  Cash through the bank                XAP             NaN   
768449   Cash through the bank                XAP   Unaccompanied   
419154   Cash through the bank                 HC   Unaccompanied   
827004                     XNA                XAP             NaN   

        NAME_CLIENT_TYPE NAME_GOODS_CATEGORY NAME_PORTFOLIO NAME_PRODUCT_TYPE  \
886608          Repeater                 XNA            XNA               XNA   
1214692        Refreshed                 XNA           Cash           walk-in   
1105368        Refreshed         Audio/Video            POS               XNA   
768449          Repeater                 XNA           Cash            x-sell   
419154          Repeater                 XNA           Cash            x-sell   
827004          Repeater                 XNA            XNA               XNA   

                    CHANNEL_TYPE


Ringkasan Statistik Data Kategorikal:


count unique                      top     freq
NAME_CONTRACT_TYPE           1670214      4               Cash loans   747553
WEEKDAY_APPR_PROCESS_START   1670214      7                  TUESDAY   255118
FLAG_LAST_APPL_PER_CONTRACT  1670214      2                        Y  1661739
NAME_CASH_LOAN_PURPOSE       1670214     25                      XAP   922661
NAME_CONTRACT_STATUS         1670214      4                 Approved  1036781
NAME_PAYMENT_TYPE            1670214      4    Cash through the bank  1033552
CODE_REJECT_REASON           1670214      9                      XAP  1353093
NAME_TYPE_SUITE               849809      7            Unaccompanied   508970
NAME_CLIENT_TYPE             1670214      4                 Repeater  1231261
NAME_GOODS_CATEGORY          1670214     28                      XNA   950809
NAME_PORTFOLIO               1670214      5                      POS   691011
NAME_PRODUCT_TYPE            1670214      3                      XNA  1063666
CHANNEL_TYPE                 1670214      8  Credit and cash offices   719968
NAME_SELLER_INDUSTRY         1670214     11                      XNA   855720
NAME_YIELD_GROUP             1670214      5                      XNA   517215
PRODUCT_COMBINATION          1669868     17                     Cash   285990


Ringkasan Statistik Data Numerik:


count          mean            std  \
SK_ID_PREV                 1670214.0  1.923089e+06  532597.958696   
SK_ID_CURR                 1670214.0  2.783572e+05  102814.823849   
AMT_ANNUITY                1297979.0  1.595512e+04   14782.137335   
AMT_APPLICATION            1670214.0  1.752339e+05  292779.762387   
AMT_CREDIT                 1670213.0  1.961140e+05  318574.616546   
AMT_DOWN_PAYMENT            774370.0  6.697402e+03   20921.495410   
AMT_GOODS_PRICE            1284699.0  2.278473e+05  315396.557937   
HOUR_APPR_PROCESS_START    1670214.0  1.248418e+01       3.334028   
NFLAG_LAST_APPL_IN_DAY     1670214.0  9.964675e-01       0.059330   
RATE_DOWN_PAYMENT           774370.0  7.963682e-02       0.107823   
RATE_INTEREST_PRIMARY         5951.0  1.883569e-01       0.087671   
RATE_INTEREST_PRIVILEGED      5951.0  7.735025e-01       0.100879   
DAYS_DECISION              1670214.0 -8.806797e+02     779.099667   
SELLERPLACE_AREA           1670214.0  3.139511e+02    7127.443459   
CNT_PAYMENT                1297984.0  1.605408e+01      14.567288   
DAYS_FIRST_DRAWING          997149.0  3.422099e+05   88916.115834   
DAYS_FIRST_DUE              997149.0  1.382627e+04   72444.869708   
DAYS_LAST_DUE_1ST_VERSION   997149.0  3.376777e+04  106857.034789   
DAYS_LAST_DUE               997149.0  7.658240e+04  149647.415123   
DAYS_TERMINATION            997149.0  8.199234e+04  153303.516729   
NFLAG_INSURED_ON_APPROVAL   997149.0  3.325702e-01       0.471134   

                                    min           25%           50%  \
SK_ID_PREV                 1.000001e+06  1.461857e+06  1.923110e+06   
SK_ID_CURR                 1.000010e+05  1.893290e+05  2.787145e+05   
AMT_ANNUITY                0.000000e+00  6.321780e+03  1.125000e+04   
AMT_APPLICATION            0.000000e+00  1.872000e+04  7.104600e+04   
AMT_CREDIT                 0.000000e+00  2.416050e+04  8.054100e+04   
AMT_DOWN_PAYMENT          -9.000000e-01  0.000000e+00  1.638000e+03   
AMT_GOODS_PRICE            0.000000e+00  5.084100e+04  1.123200e+05   
HOUR_APPR_PROCESS_START    0.000000e+00  1.000000e+01  1.200000e+01   
NFLAG_LAST_APPL_IN_DAY     0.000000e+00  1.000000e+00  1.000000e+00   
RATE_DOWN_PAYMENT         -1.497876e-05  0.000000e+00  5.160508e-02   
RATE_INTEREST_PRIMARY      3.478125e-02  1.607163e-01  1.891222e-01   
RATE_INTEREST_PRIVILEGED   3.731501e-01  7.156448e-01  8.350951e-01   
DAYS_DECISION             -2.922000e+03 -1.300000e+03 -5.810000e+02   
SELLERPLACE_AREA          -1.000000e+00 -1.000000e+00  3.000000e+00   
CNT_PAYMENT                0.000000e+00  6.000000e+00  1.200000e+01   
DAYS_FIRST_DRAWING        -2.922000e+03  3.652430e+05  3.652430e+05   
DAYS_FIRST_DUE            -2.892000e+03 -1.628000e+03 -8.310000e+02   
DAYS_LAST_DUE_1ST_VERSION -2.801000e+03 -1.242000e+03 -3.610000e+02   
DAYS_LAST_DUE             -2.889000e+03 -1.314000e+03 -5.370000e+02   
DAYS_TERMINATION          -2.874000e+03 -1.270000e+03 -4.990000e+02   
NFLAG_INSURED_ON_APPROVAL  0.000000e+00  0.000000e+00  0.000000e+00   

                                    75%          max  
SK_ID_PREV                 2.384280e+06  2845382.000  
SK_ID_CURR                 3.675140e+05   456255.000  
AMT_ANNUITY                2.065842e+04   418058.145  
AMT_APPLICATION            1.803600e+05  6905160.000  
AMT_CREDIT                 2.164185e+05  6905160.000  
AMT_DOWN_PAYMENT           7.740000e+03  3060045.000  
AMT_GOODS_PRICE            2.340000e+05  6905160.000  
HOUR_APPR_PROCESS_START    1.500000e+01       23.000  
NFLAG_LAST_APPL_IN_DAY     1.000000e+00        1.000  
RATE_DOWN_PAYMENT          1.089091e-01        1.000  
RATE_INTEREST_PRIMARY      1.933299e-01        1.000  
RATE_INTEREST_PRIVILEGED   8.525370e-01        1.000  
DAYS_DECISION             -2.800000e+02       -1.000  
SELLERPLACE_AREA           8.200000e+01  4000000.000  
CNT_PAYMENT                2.400000e+01       84.000  
DAYS_FIRST_DRAWING         3.652430e+05   365243.000  
DAYS_FIRST_DUE

In [6]:
print("Categorical columns: ", categorical_col)
print("Numerical columns: ", numerical_col)

Categorical columns:  ['NAME_CONTRACT_TYPE', 'WEEKDAY_APPR_PROCESS_START', 'FLAG_LAST_APPL_PER_CONTRACT', 'NAME_CASH_LOAN_PURPOSE', 'NAME_CONTRACT_STATUS', 'NAME_PAYMENT_TYPE', 'CODE_REJECT_REASON', 'NAME_TYPE_SUITE', 'NAME_CLIENT_TYPE', 'NAME_GOODS_CATEGORY', 'NAME_PORTFOLIO', 'NAME_PRODUCT_TYPE', 'CHANNEL_TYPE', 'NAME_SELLER_INDUSTRY', 'NAME_YIELD_GROUP', 'PRODUCT_COMBINATION']
Numerical columns:  ['SK_ID_PREV', 'SK_ID_CURR', 'AMT_ANNUITY', 'AMT_APPLICATION', 'AMT_CREDIT', 'AMT_DOWN_PAYMENT', 'AMT_GOODS_PRICE', 'HOUR_APPR_PROCESS_START', 'NFLAG_LAST_APPL_IN_DAY', 'RATE_DOWN_PAYMENT', 'RATE_INTEREST_PRIMARY', 'RATE_INTEREST_PRIVILEGED', 'DAYS_DECISION', 'SELLERPLACE_AREA', 'CNT_PAYMENT', 'DAYS_FIRST_DRAWING', 'DAYS_FIRST_DUE', 'DAYS_LAST_DUE_1ST_VERSION', 'DAYS_LAST_DUE', 'DAYS_TERMINATION', 'NFLAG_INSURED_ON_APPROVAL']


In [7]:
dfe = df.copy()
# Memeriksa baris duplikat
duplicate_rows = dfe.duplicated()

# Menampilkan jumlah baris duplikat
print(f"Jumlah baris duplikat: {duplicate_rows.sum()}")

# Fungsi untuk menghitung dan menampilkan kolom dengan nilai missing
def check_nan(dfe):
    # Hitung nilai null
    null_values = dfe.isnull().sum()

    # Buat DataFrame untuk total nilai null dan persentase
    nvc = pd.DataFrame(null_values[null_values > 0].sort_values(), columns=['Total Null Values'])  # Hanya ambil kolom dengan nilai null
    nvc['Percentage'] = (nvc['Total Null Values'] / dfe.shape[0]) * 100
    nvc["Data Type"] = [dfe[col].dtype for col in nvc.index]  # Ambil tipe data dari kolom yang ada

    # Urutkan berdasarkan total nilai null dan persentase
    nvc.sort_values(by=["Total Null Values", "Percentage"], ascending=False, inplace=True)

    # Tampilkan hasil dengan gaya
    return nvc.style.background_gradient(cmap='Blues')

# Panggil fungsi dan simpan hasilnya
styled_result = check_nan(dfe)

# Untuk menampilkan hasil di Jupyter Notebook, gunakan:
styled_result

Jumlah baris duplikat: 0


In [8]:
# 1. Drop kolom dengan missing value > 99%
dfe.drop(columns=['RATE_INTEREST_PRIMARY', 'RATE_INTEREST_PRIVILEGED'], inplace=True)

# 2. Imputasi numerik
dfe['AMT_DOWN_PAYMENT'].fillna(0, inplace=True)
dfe['RATE_DOWN_PAYMENT'].fillna(0, inplace=True)
dfe['DAYS_FIRST_DRAWING'].fillna(dfe['DAYS_FIRST_DRAWING'].median(), inplace=True)
dfe['DAYS_FIRST_DUE'].fillna(dfe['DAYS_FIRST_DUE'].median(), inplace=True)
dfe['DAYS_LAST_DUE_1ST_VERSION'].fillna(dfe['DAYS_LAST_DUE_1ST_VERSION'].median(), inplace=True)
dfe['DAYS_LAST_DUE'].fillna(dfe['DAYS_LAST_DUE'].median(), inplace=True)
dfe['DAYS_TERMINATION'].fillna(dfe['DAYS_TERMINATION'].median(), inplace=True)
dfe['NFLAG_INSURED_ON_APPROVAL'].fillna(0, inplace=True)
dfe['AMT_GOODS_PRICE'].fillna(dfe['AMT_GOODS_PRICE'].median(), inplace=True)
dfe['AMT_ANNUITY'].fillna(dfe['AMT_ANNUITY'].median(), inplace=True)
dfe['CNT_PAYMENT'].fillna(dfe['CNT_PAYMENT'].median(), inplace=True)
dfe['AMT_CREDIT'].fillna(dfe['AMT_CREDIT'].median(), inplace=True)

# 3. Imputasi kategorikal
dfe['NAME_TYPE_SUITE'].fillna('Unknown', inplace=True)
dfe['PRODUCT_COMBINATION'].fillna(dfe['PRODUCT_COMBINATION'].mode()[0], inplace=True)


AMT_DOWN_PAYMENT, 53.6%	Bisa imputasi dengan 0 (banyak kredit tanpa uang muka), atau median

RATE_DOWN_PAYMENT, 53.6% Imputasi dengan 0 masuk akal (karena korelasi kuat dgn AMT_DOWN_PAYMENT)

NAME_TYPE_SUITE, 49.1%, Kategorikal → isi dengan "Unknown" atau mode

DAYS_FIRST_DRAWING, 40.3%, Imputasi dengan median (atau nilai default = 365243 jika implikasinya valid)

DAYS_FIRST_DUE, 40.3%, Imputasi dengan median

DAYS_LAST_DUE_1ST_VERSION, 40.3%, Imputasi dengan median

DAYS_LAST_DUE, 40.3%, Imputasi dengan median

DAYS_TERMINATION, 40.3%, Imputasi dengan median

NFLAG_INSURED_ON_APPROVAL, 40.3%, Imputasi dengan 0 (artinya tidak diasuransikan → default)

AMT_GOODS_PRICE, 23.1%, Imputasi dengan median

AMT_ANNUITY, 22.3%, Imputasi dengan median

CNT_PAYMENT, 22.3%, Imputasi dengan median

PRODUCT_COMBINATION	0.02%, Kategorikal = imputasi dengan mode

AMT_CREDIT, 0.00006%, Satu baris saja = imputasi dengan median


In [9]:
# Panggil fungsi dan simpan hasilnya
styled_result = check_nan(dfe)

# Untuk menampilkan hasil di Jupyter Notebook, gunakan:
styled_result

In [10]:
dfe.sample(15)

SK_ID_PREV  SK_ID_CURR NAME_CONTRACT_TYPE  AMT_ANNUITY  \
1495219     1722348      146979    Revolving loans    11250.000   
1159725     1934806      121484         Cash loans    11250.000   
672430      2476135      192083         Cash loans    15955.560   
49903       2572755      151900         Cash loans    25198.650   
493305      2615587      168175         Cash loans    11250.000   
1162669     1544717      370338     Consumer loans     7957.305   
1632470     2821188      390777     Consumer loans     3038.175   
845511      2703709      310389     Consumer loans     6901.110   
69211       1546606      144283     Consumer loans     6472.620   
1083608     1292607      294290     Consumer loans     7773.885   
846807      2259728      239722    Revolving loans     2250.000   
1529102     1815086      204229         Cash loans    22017.915   
1377711     2718998      343617     Consumer loans     5301.810   
97116       2064963      152689         Cash loans    11250.000   
1129779     2817869      415355     Consumer loans     9943.920   

         AMT_APPLICATION  AMT_CREDIT  AMT_DOWN_PAYMENT  AMT_GOODS_PRICE  \
1495219              0.0         0.0               0.0         112320.0   
1159725              0.0         0.0               0.0         112320.0   
672430          238500.0    299466.0               0.0         238500.0   
49903           225000.0    239850.0               0.0         225000.0   
493305               0.0         0.0               0.0         112320.0   
1162669          63000.0     42507.0           22500.0          63000.0   
1632470          27846.0     22266.0            5580.0          27846.0   
845511           73935.0     73935.0               0.0          73935.0   
69211            33628.5     35626.5               0.0          33628.5   
1083608          73885.5     71451.0            8869.5          73885.5   
846807           45000.0     45000.0               0.0          45000.0   
1529102         193500.0    206271.0               0.0         193500.0   
1377711         129474.0    116523.0           12951.0         129474.0   
97116                0.0         0.0               0.0         112320.0   
1129779          56011.5     53068.5            5602.5          56011.5   

        WEEKDAY_APPR_PROCESS_START  HOUR_APPR_PROCESS_START  \
1495219                     FRIDAY                       18   
1159725                  WEDNESDAY                       10   
672430                      FRIDAY                       10   
49903                      TUESDAY                       14   
493305                      FRIDAY                       18   
1162669                     SUNDAY                        9   
1632470                   SATURDAY                       16   
845511                    SATURDAY                       12   
69211                    WEDNESDAY                       17   
1083608                  WEDNESDAY                       14   
846807                   WEDNESDAY                       14   
1529102                     SUNDAY                       11   
1377711                     FRIDAY                       19   
97116                     THURSDAY                       18   
1129779                  WEDNESDAY                       15   

        FLAG_LAST_APPL_PER_CONTRACT  NFLAG_LAST_APPL_IN_DAY  \
1495219                           Y                       1   
1159725                           Y                       1   
672430                            Y                       1   
49903                             Y                       1   
493305                            Y                       1   
1162669                           Y                       1   
1632470                           Y                       1   
845511                            Y                       1   
69211                             Y                       1   
1083608                           Y                       1   
846807                           

In [11]:
dfe.to_csv('prev_app_cleaned.csv', index=False)